kroxylicious-docs/docs/_modules/record-encryption/azure-key-vault/con-azure-key-vault-key-creation.adoc

```diff
@@ -0,0 +1,51 @@
+:_mod-docs-content-type: CONCEPT
+
+// file included in the following:
+//
+// assembly-azure-key-vault.adoc
+
+[id='con-azure-key-vault-key-creation-{context}']
+= Creating Azure Key Vault keys
+
+[role="_abstract"]
+As a user with the *Key Vault Crypto Officer* role (or otherwise sufficient {microsoft-learn}/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations[RBAC] or {microsoft-learn}/azure/key-vault/general/rbac-access-policy[legacy] permissions to manage keys in Azure Key Vault), use either the {microsoft-learn}/cli/azure/keyvault/key?view=azure-cli-latest#az-keyvault-key-create[Azure CLI] or {microsoft-learn}/powershell/module/az.keyvault/add-azkeyvaultkey[Azure PowerShell] to create a *key* with *wrapKey* and *unwrapKey* operations enabled.
+
+== Establish the naming convention for keys within key vault hierarchy
+
+Establish a naming convention for keys to keep the filter’s keys separate from those used by other systems.
+
+NOTE: The filter key name prefix `KEK_` is used widely across the Kroxylicious project, including here in the documentation, as well as in the quickstarts, developer guides, examples, and scripts. However, *Azure Key Vault does not permit underscore characters in key names*, so Azure Key Vault users will have to adjust this naming scheme as required.
+
+== Supported Key Types
+
+NOTE: The key types available to you in Azure Key Vault will depend on your Azure Key Vault service tier. See the {microsoft-learn}/azure/key-vault/keys/about-keys[_About keys_ page in the Azure Key Vault docs] for more details.
```

Comment:
I might say here:

NOTE: The key types available in Azure Key Vault depend on your service tier. For current availability and restrictions, see <<Microsoft Learn’s Azure Key Vault documentation>>.  (https://learn.microsoft.com/en-us/azure/key-vault/)

I think there are too many links sending the reader to sections in MicroSoft Learn.
Distracting if you are familiar with the setup.
Maybe reduce and have an additional resources at the start of the assembly instead of lots of embedded links:

[role="_additional-resources"]
.Additional resources

* Azure Key Vault documentation ({microsoft-learn}/azure/key-vault/)